cmd/root.go

```diff
@@ -33,12 +35,29 @@ var rootTokenFromFlag string
 // by main.main(). It only needs to happen once to
 // the rootCmd.
 func Execute() {
+	header.SetCommandStr(CommandStr())
 	command := MakeCommands()
 	if err := command.Execute(); err != nil {
 		os.Exit(-1)
 	}
 }
 
+// Returns a string (e.g. "context list") indicating what
+// subcommand is being called, without any args or flags,
+// for API headers.
+func CommandStr() string {
+	command := MakeCommands()
+	subCmd, _, err := command.Find(os.Args[1:])
+	if err != nil {
+		return ""
+	}
+	parentNames := []string{subCmd.Name()}
+	subCmd.VisitParents(func(parent *cobra.Command) {
+		parentNames = append([]string{parent.Name()}, parentNames...)
```

Comment:
there's gotta be a better way to build up a slice of strings backwards, right?

-- a Go noob